Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -69,7 +70,7 @@ class ActivityStreamDataObserver: DataObserver {
         }
 
         self.delegate?.willInvalidateDataSources()
-        self.profile.recommendations.repopulate(invalidateTopSites: shouldInvalidateTopSites, invalidateHighlights: shouldInvalidateHighlights).uponQueue(.main) { _ in
+        self.profile.recommendations.repopulate(invalidateTopSites: shouldInvalidateTopSites, invalidateHighlights: shouldInvalidateHighlights).uponQueue(queue) { _ in
```

Comment:
`uponQueue` means "when all this work is done, please run this callback on the provided queue".

What we want to do here is do the repopulation on the non-main queue, and continue to run this callback — which does datasource stuff — on the main queue, no?